### PR TITLE
Update Ceph troubleshooting guide

### PR DESCRIPTION
**Pull request by fmount**
This patch updates the `Ceph` troubleshooting guide to use a debug (`Glance`) `Pod` to run `rbd` commands against the `Ceph` cluster. `Glance` is connected to the storage network through the proper `NAD`, and it's a good fit to provide a `Ceph` troubleshooting example.

Jira: [OSPRH-839](https://issues.redhat.com/browse/OSPRH-839)

### CODE REVIEW

File: troubleshooting.md [hunk 1]
```diff
@@ -118,28 +118,38 @@ troubleshooting guide.
...
-ports of the Ceph cluster's monitors should be in
-`/etc/ceph.conf`. From within the pod, read this and try to ping the
-IP as well as connect to the port.
+The Ceph cluster should be reachable from the Storage network as described in
+the [networking document](networking.md). The IPs and ports of the Ceph
+cluster's monitors should be in `/etc/ceph.conf`. From within the pod, read
+this and try to reach the IP as well as connect to the port.
 ```
-$ oc rsh -c glance-api glance-default-external-api-0
+$ oc debug pod/glance-default-external-api-0
```
Comment (fultonj) [inline troubleshooting.md:152]: I think it should be `oc debug --container glance-api pod/glance-default-external-api-0` instead.

When I use `--container glance-api` then I can see the relevant files.
```
[zuul@controller-0 ~]$ oc debug --container glance-api pod/glance-3d293-default-external-api-0 
Warning: would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (containers "glance-log", "glance-httpd", "glance-api" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (containers "glance-log", "glance-httpd", "glance-api" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or containers "glance-log", "glance-httpd", "glance-api" must set securityContext.runAsNonRoot=true), runAsUser=0 (containers "glance-log", "glance-httpd", "glance-api" must not set runAsUser=0), seccompProfile (pod or containers "glance-log", "glance-httpd", "glance-api" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
Starting pod/glance-3d293-default-external-api-0-debug-sbtgs, command was: /usr/bin/dumb-init --single-child -- /bin/bash -c /usr/local/bin/kolla_set_configs && /usr/local/bin/kolla_start
Pod IP: 192.168.23.132
If you don't see a command prompt, try pressing enter.
sh-5.1# ls -l /etc/ceph/
total 0
lrwxrwxrwx. 1 root root 36 Jul 29 13:09 ceph.client.openstack.keyring -> ..data/ceph.client.openstack.keyring
lrwxrwxrwx. 1 root root 16 Jul 29 13:09 ceph.conf -> ..data/ceph.conf
sh-5.1# 
```
But without it, then I cannot see the ceph.conf in my debug pod.
```
[zuul@controller-0 ~]$ oc get pods | grep glance
glance-3d293-default-external-api-0                               3/3     Running     0             47h
glance-3d293-default-external-api-1                               3/3     Running     0             47h
glance-3d293-default-external-api-2                               3/3     Running     0             47h
glance-3d293-default-internal-api-0                               3/3     Running     0             47h
glance-3d293-default-internal-api-1                               3/3     Running     0             47h
glance-3d293-default-internal-api-2                               3/3     Running     0             47h
glance-dbpurge-28700641-6prqz                                     0/1     Completed   0             2d13h
glance-dbpurge-28702081-g5xlx                                     0/1     Completed   0             37h
glance-dbpurge-28703521-rd5z9                                     0/1     Completed   0             13h
[zuul@controller-0 ~]$ oc debug pod/glance-3d293-default-external-api-0
Defaulting container name to glance-log.
Use 'oc describe pod/glance-3d293-default-external-api-0-debug-tqjjq -n openstack' to see all of the containers in this pod.

Warning: would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (containers "glance-log", "glance-httpd", "glance-api" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (containers "glance-log", "glance-httpd", "glance-api" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or containers "glance-log", "glance-httpd", "glance-api" must set securityContext.runAsNonRoot=true), runAsUser=0 (containers "glance-log", "glance-httpd", "glance-api" must not set runAsUser=0), seccompProfile (pod or containers "glance-log", "glance-httpd", "glance-api" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
Starting pod/glance-3d293-default-external-api-0-debug-tqjjq, command was: /usr/bin/dumb-init --single-child -- /bin/sh -c /usr/bin/tail -n+1 -F /var/log/glance/glance-3d293-default-external.log 2>/dev/null
Pod IP: 192.168.23.131
If you don't see a command prompt, try pressing enter.
sh-5.1# ls -l /etc/ceph/
total 4
-rw-r--r--. 1 root root 92 Apr 29 21:07 rbdmap
sh-5.1# 
```

File: troubleshooting.md [hunk 2]
```diff
@@ -156,32 +166,11 @@ If you are unable to connect, then troubleshoot the network.
...
-The above update will restart the NovaAPI pod. Connect to the
-restarted pod using `oc rsh` and confirm the required keys
-are present as described in step 1.
-
-Once connected to the NovaAPI pod with `oc rsh` examine the cephx key.
+`rbd` command to read from it.
+
+**Note**
+
+Before running the rbd command, examine the cephx key.
```
Comment (fultonj) [inline troubleshooting.md:173]: The command below on line 189 and 190 needs back slashes.

When I run it as written it doesn't parse.

Also on line 222 please remove the `^C`.

**Review by fultonj — COMMENTED** [review]
Overall this is a good update so thank you Francesco. 

Just requesting some small changes.

File: troubleshooting.md (hunk 1 above)
Comment (fmount) [reply troubleshooting.md:152]: +1 good catch! Updating accordingly!

File: troubleshooting.md (hunk 2 above)
Comment (fmount) [reply troubleshooting.md:173]: Done

File: troubleshooting.md (hunk 1 above)
Comment (fmount) [reply troubleshooting.md]: Done (thank you!)

**Pull request merged**